### PR TITLE
Only remove quote if regex not found

### PR DESCRIPTION
**Pull request by evonck**
This PR is trying to fix bug link to adding a quote messages in your reply.

Instead of starting by removing the quote from gmail, microsoft etc... We first attempt to remove the quote based on the regex. If no regex is found then use the blockquote.

This PR is increasing the minimum node and lines, as we don't remove the quote before checking the marked line we need to make sure the limit is high enough to check most of the emails.

New PR to set this as a param will come after this one.

### CODE REVIEW

**Comment by evonck** [comment]
@quentez updated PR, if you have time to look at this one it would be great. Thanks

File: tests/fixtures/nylas/email_1_stripped.html [hunk 1]
```diff
@@ -1 +1,17 @@
...
+      <pre
+        class="special_formatting"><font face="Calibri"><span style="font-size: 15px;">Hi Jeff,  Quick update on the event bugs: - I fixed the bug where events would be incorrectly marked as read-only. - We expose RRULEs as valid JSON now.   We're currently testing the fixes, they should ship early next week.  Concerning timezones, an event should always be associated with a timezone. Having a NULL value instead is a bug on our end. I will be working on fixing this on Monday and will let you know when it's fixed.  Thanks for your detailed bug reports,</span></font></pre>
+      <pre class="special_formatting"><font face="Calibri"><span style="font-size: 15px;"> Karim</span></font></pre>
+      <pre class="special_formatting"
+        style="color: rgb(0, 0, 0); font-family: Calibri, sans-serif; font-size: 14px;"><span style="font-family: Calibri; font-size: 11pt; font-weight: bold;">From: </span><span style="font-family: Calibri; font-size: 11pt;"> Kavya Joshi &lt;</span><a href="mailto:kavya@nylas.com" style="font-family: Calibri; font-size: 11pt;">kavya@nylas.com</a><span style="font-family: Calibri; font-size: 11pt;">></span></pre>
+    </span></div>
```
Comment (evonck) [inline tests/fixtures/nylas/email_1_stripped.html:14]: Remove `<span id="OLK_SRC_BODY_SECTION" style="color: rgb(0, 0, 0); font-family: Calibri, sans-serif; font-size: 14px;"> </span>` (This was check with product.)

File: tests/fixtures/nylas/email_16_stripped.html [hunk 2]
```diff
@@ -1 +1,198 @@
...
+
+<head>
+  <meta http-equiv="Content-Type" content="text/html; charset=us-ascii" />
+</head>
+
+<body>
+  <blockquote class="gmail_quote" style="margin:0 0 0 .8ex;border-left:1px #ccc solid;padding-left:1ex;">
+    <div>Hi all,</div>
+    <div><br /> </div>
+    <div>Below is the sign up link for on-site massages tomorrow. The moussuse will arrive after lunch. Please sign up
```
Comment (evonck) [inline tests/fixtures/nylas/email_16_stripped.html:11]: Add 
`Below is the sign up link for on-site massages tomorrow. The moussuse will arrive after lunch. Please sign up for your time if you wish to participate :)` Keep the style

File: tests/fixtures/nylas/email_12_stripped.html [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
-            <br/> See also
-            <br/> http://wiki.basho.com/REST-API.html#Bucket-operations
-            <br/> which says
-            <br/> "At the moment there is no straightforward way to delete an entire Bucket. There is, however, an open ticket for the feature. To delete all the keys in a bucket, you’ll need to delete them all individually." </p>
-        <br/>
-        <br/>
-        <br/>
-        <br/> _______________________________________________ riak-users mailing list riak-users@lists.basho.com http://lists.basho.com/mailman/listinfo/riak-users_lists.basho.com </div>
+    <div> Hi,<br />
+        <blockquote class="foo"> On Tue, 2011-03-01 at 18:02 +0530, Abhishek Kona wrote: <div> > Hi folks <br /> >
```
Comment (evonck) [inline tests/fixtures/nylas/email_12_stripped.html:5]: Keep the quote, as we answer the email

**Comment by quentez** [comment]
Why is there a `#20` in the title?

File: src/Quotations.ts [hunk 4]
```diff
@@ -121,20 +115,34 @@ export function extractFromHtml(messageBody: string): ExtractFromHtmlResult {
...
   // Otherwise, if we found a known quote earlier, return the content before.
-  else if (!extractQuoteHtml.quoteWasFound && cutQuotations)
+  if (cutQuotations)
     return { body: xmlDomSerializer.serializeToString(xmlDocumentCopy, true), didFindQuote: true };
   // Finally, if no quote was found, return the original HTML.
   else
     return { body: messageBody, didFindQuote: false };
 }
 
+function cutQuotation(xmlDocument: Document, options?: cutQuoteOption) {
```
Comment (quentez) [inline src/Quotations.ts]: Types should be uppercase.

```suggestion
function cutQuotation(xmlDocument: Document, options?: CutQuoteOption) {
```

File: src/Quotations.ts [hunk 5]
```diff
@@ -239,7 +247,6 @@ export function markMessageLines(lines: string[]): string {
   let index = 0;
   while (index < lines.length) {
     const line = lines[index];
-
```
Comment (quentez) [inline src/Quotations.ts]: Why the change in spacing?

File: src/Quotations.ts [hunk 6]
```diff
@@ -121,20 +115,34 @@ export function extractFromHtml(messageBody: string): ExtractFromHtmlResult {
...
     // Serialize and return.
     return {
       body: xmlDomSerializer.serializeToString(xmlDocumentCopy, true),
       didFindQuote: true
     }
   }
+   // Try and cut the quote of one of the known types.
+   const cutQuotations = cutQuotation(xmlDocumentCopy);
+
   // Otherwise, if we found a known quote earlier, return the content before.
```
Comment (quentez) [inline src/Quotations.ts]: We may want to update the comments here.

File: src/HtmlQuotations.ts [hunk 7]
```diff
@@ -112,33 +111,41 @@ export function deleteQuotationTags(document: Document, element: Node, quotation
     isTagInQuotation
   };
 }
+export interface cutQuoteOption {
```
Comment (quentez) [inline src/HtmlQuotations.ts]: Types should be uppercase.
Also plural in this case.

```suggestion
export interface CutQuoteOptions {
```

File: src/HtmlQuotations.ts [hunk 8]
```diff
@@ -58,7 +58,6 @@ export function deleteQuotationTags(document: Document, element: Node, quotation
   isTagInQuotation: boolean
 } {
   let isTagInQuotation = true;
-
```
Comment (quentez) [inline src/HtmlQuotations.ts]: ?

File: src/HtmlQuotations.ts [hunk 9]
```diff
@@ -112,33 +111,41 @@ export function deleteQuotationTags(document: Document, element: Node, quotation
...
+  if (!gmailQuote || (gmailQuote.textContent && matchStart(gmailQuote.textContent, ForwardRegexp)) || shouldNotRemoveQuote(gmailQuote, options))
     return false;
 
   // Otherwise, remove the quote from the document and return.
   gmailQuote.parentNode.removeChild(gmailQuote);
   return true;
 };
 
+function shouldNotRemoveQuote(node: Node, options?: cutQuoteOption) {
+  return options && options.onlyRemoveEmptyBlocks && node.textContent.trim() !== '';
```
Comment (quentez) [inline src/HtmlQuotations.ts]: We could cast to boolean here just to have a clean return type.

File: src/HtmlQuotations.ts [hunk 10]
```diff
@@ -112,33 +111,41 @@ export function deleteQuotationTags(document: Document, element: Node, quotation
...
-  if (!gmailQuote || (gmailQuote.textContent && matchStart(gmailQuote.textContent, ForwardRegexp)))
+  if (!gmailQuote || (gmailQuote.textContent && matchStart(gmailQuote.textContent, ForwardRegexp)) || shouldNotRemoveQuote(gmailQuote, options))
     return false;
 
   // Otherwise, remove the quote from the document and return.
   gmailQuote.parentNode.removeChild(gmailQuote);
   return true;
 };
 
+function shouldNotRemoveQuote(node: Node, options?: cutQuoteOption) {
```
Comment (quentez) [inline src/HtmlQuotations.ts]: This is used by functions both above and below.
I feel like it should be either at the beginning or at the end.

File: src/HtmlQuotations.ts [hunk 11]
```diff
@@ -112,33 +111,41 @@ export function deleteQuotationTags(document: Document, element: Node, quotation
     isTagInQuotation
   };
 }
```
Comment (quentez) [inline src/HtmlQuotations.ts:114]: Spacing.

File: src/HtmlQuotations.ts [hunk 12]
```diff
@@ -230,65 +239,19 @@ export function cutById(document: Document): boolean {
...
   return true;
-};
-
-/**
- * Cuts div tag that wraps a block starting with "From:".
- *
- * @param {Document} document - The document to cut the element from.
- * @return {boolean} Whether a corresponding quote was found or not.
- */
-export function cutFromBlock(document: Document): boolean {
```
Comment (quentez) [inline src/HtmlQuotations.ts:255]: Was that just not used anywhere anymore?

**Review by quentez — CHANGES_REQUESTED** [review]

File: src/HtmlQuotations.ts (hunk 12 above)
Comment (evonck) [reply src/HtmlQuotations.ts:255]: Looks like it was remove in thie PR : https://github.com/quentez/talonjs/commit/191d79d26c43eac297abd19fbdef796e37ece3f8

**Review by quentez — APPROVED** [review]
